.github/workflows/publish-dist.yml

```diff
@@ -8,57 +8,34 @@ jobs:
   run:
     if: github.event.workflow_run.conclusion == 'success'
     runs-on: ubuntu-latest
+    timeout-minutes: 10
     steps:
       - name: Download build-dist artifacts
         uses: dawidd6/action-download-artifact@v2
         with:
+          name: dist
           workflow: build-dist
           run_id: ${{ github.event.workflow_run.id }}
+          path: dist-repo
 
       - name: Set up configuration and secrets
         run: |
           printf '[user]\n\tname = Cockpit Project\n\temail=cockpituous@gmail.com\n' > ~/.gitconfig
-          echo '${{ secrets.COCKPITUOUS_TOKEN }}' > ~/.config/github-token
           # we push to -dist repo via https://github.com, that needs our cockpituous token
           git config --global credential.helper store
           echo 'https://token:${{ secrets.COCKPITUOUS_TOKEN }}@github.com' >> ~/.git-credentials
 
-      - name: Commit dist tarball dist repo
+      - name: Commit to dist repo
         run: |
           set -ex
+          SHA="$(cat dist-repo/SHA)"
```

Comment:
This is insecure.

The SHA came from a workflow which is under complete control of a potentially-hostile pull request.  Someone could open a pull request against cockpit, populate our current master branch SHA into this file, and then we'd pick it up here and tag their malicious payload as a valid cache of the current state of our master branch.

This is what I've been spending my week worrying about... I think we need to do better here.

One approach might be to upload things to git only after stuff lands on master, and use some form of artifact-based downloading for testing pull requests...

Another approach might be to use `pull_request_target` and try to come up with a way to harden the build.  For the `node_modules` stuff I'm running in a separate docker container, for example...

Reply:
Of course it is untrusted, that's what the current master workflow has as well. But that's fine -- us project mainainers must review a PR. It's just a cache, it's exactly the same as locally checking out and building an unreviewed PR.

Reply:
Discussed in meeting. Will rework this, in a separate PR. Either with:
 * Not storing the SHA in the workflow, but instead querying the github.event.workflow for the originally proposed repo/branch and determining the SHA that way.
 * If that somehow does not work out, restrict publish-dist to PRs whose owners are in cockpit-project. We'd lose packit tests for "outsider" PRs, but that does not hurt that much. (At least until we figure out a better way)

Reply:
Fixed in separate commit, similar to https://github.com/cockpit-project/cockpit-machines/pull/143